examples/03_second_level_models/plot_oasis.py

```diff
@@ -0,0 +1,115 @@
+"""Voxel-Based Morphometry on Oasis dataset
+========================================
+
+This example uses Voxel-Based Morphometry (VBM) to study the relationship
+between aging, sex and gray matter density.
+
+The data come from the `OASIS <http://www.oasis-brains.org/>`_ project.
+If you use it, you need to agree with the data usage agreement available
+on the website.
+
+It has been run through a standard VBM pipeline (using SPM8 and
+NewSegment) to create VBM maps, which we study here.
+
+VBM analysis of aging
+---------------------
+
+We run a standard GLM analysis to study the association between age
+and gray matter density from the VBM data. We use only 100 subjects
+from the OASIS dataset to limit the memory usage.
+
+Note that more power would be obtained from using a larger sample of subjects.
+____
+
+"""
+# Authors: Bertrand Thirion, <bertrand.thirion@inria.fr>, July 2018
+#          Elvis Dhomatob, <elvis.dohmatob@inria.fr>, Apr. 2014
+#          Virgile Fritsch, <virgile.fritsch@inria.fr>, Apr 2014
+#          Gael Varoquaux, Apr 2014
+import numpy as np
+import matplotlib.pyplot as plt
+from nilearn import datasets
+from nilearn.input_data import NiftiMasker
+
+n_subjects = 100  # more subjects requires more memory
+
+############################################################################
+# Load Oasis dataset
+# -------------------
+oasis_dataset = datasets.fetch_oasis_vbm(n_subjects=n_subjects)
+gray_matter_map_filenames = oasis_dataset.gray_matter_maps
```

Comment:
@bthirion I think that a pedagogic issue with this is that it's unclear how to assemble a dataset to be analysed by `nistats`. 

* Do I always need to save all the images on my dataset and then give it a list of filenames?
* Can I give it a 4-dimensional image? If yes, over what dimension are we performing the analysis?
* Can I give it a list of images?

Reply:
You can give either a 4-dimensional image or a list of 3D images.
You can give either an already-loaded  NiftiImage(s) or a path to such image(s).
I can't figure out any difficulty here.

Reply:
This doesn't seem to work
```
from glob import glob
import nibabel as nib
import pandas
from nistats.second_level_model import SecondLevelModel
import numpy as np

fs = glob('*.nii.gz') # Assuming all images have the same dimensions

data = nib.spatialimages.SpatialImage(
    np.concatenate([nib.load(f).get_data()[..., None] for f in fs], axis=-1),
    nib.load(fs[0]).affine
)
second_level_model = SecondLevelModel(smoothing_fwhm=2.0)

c = pandas.DataFrame([[1]] * len(fs), columns=['intercept'])

r1 = second_level_model.fit(ddata, design_matrix=c)
```

Failing with
`ValueError: second_level_input must be a list of 'FirstLevelModel' objects, a pandas DataFrame or a list Niimg-like objects. Instead <class 'nibabel.nifti1.Nifti1Image'> was provided`

Also tried
```
data = nib.spatialimages.Nifti1Image(
    np.concatenate([nib.load(f).get_data()[..., None, None] for f in fs], axis=-1),
    nib.load(fs[0]).affine
)
```
which is more inline with [nifti specs](https://nifti.nimh.nih.gov/pub/dist/src/niftilib/nifti1.h) (4-th axis is temporal and 5th vector-valued data)

Reply:
`second_level_model.fit(fs, design_matrix=c)` works.
However, I agree that in the current state, the code is not flexible enough to accept a 4D image.

Reply:
Yes! There is a way to make it work. Its just that the 4D image is a common application case. Either turning it onto a list of images or having to save each image separately seems not too ergonomic to me.

Reply:
Besides  4D image aspect that will be handled in another PR, can we consider merging this PR ?